docs/src/modules/components/Demo.js

```diff
@@ -554,46 +617,81 @@ export default function Demo(props) {
               </React.Suspense>
             </NoSsr>
           </DemoToolbarRoot>
-          <Collapse in={openDemoSource} unmountOnExit timeout={150}>
-            {/* A limitation from https://github.com/nihgwu/react-runner,
+          <Tabs defaultValue={0} value={activeTab} onChange={handleTabChange}>
+            {demoData.relativeModules && openDemoSource && !editorCode.isPreview ? (
+              <CodeTabList sx={bordersOverride} ownerState={ownerState}>
+                {tabs.map((tab, index) => (
+                  <CodeTab
+                    sx={selectionOverride}
+                    ownerState={ownerState}
+                    key={tab.module}
+                    value={index}
+                  >
+                    {tab.module}
+                  </CodeTab>
+                ))}
+              </CodeTabList>
+            ) : null}
+            <Collapse in={openDemoSource} unmountOnExit timeout={150}>
+              {/* A limitation from https://github.com/nihgwu/react-runner,
                 we can't inject the `window` of the iframe so we need a disableLiveEdit option. */}
-            {demoOptions.disableLiveEdit ? (
-              <DemoCodeViewer
-                code={editorCode.value}
-                id={demoSourceId}
-                language={demoData.sourceLanguage}
-                copyButtonProps={{
-                  'data-ga-event-category': codeOpen ? 'demo-expand' : 'demo',
-                  'data-ga-event-label': demo.gaLabel,
-                  'data-ga-event-action': 'copy-click',
-                }}
-              />
-            ) : (
-              <DemoEditor
-                // Mount a new text editor when the preview mode change to reset the undo/redo history.
-                key={editorCode.isPreview}
-                value={editorCode.value}
-                onChange={(value) => {
-                  setEditorCode({
-                    ...editorCode,
-                    value,
-                  });
-                }}
-                onFocus={() => {
-                  setLiveDemoActive(true);
-                }}
-                id={demoSourceId}
-                language={demoData.sourceLanguage}
-                copyButtonProps={{
-                  'data-ga-event-category': codeOpen ? 'demo-expand' : 'demo',
-                  'data-ga-event-label': demo.gaLabel,
-                  'data-ga-event-action': 'copy-click',
-                }}
-              >
-                <DemoEditorError>{debouncedError}</DemoEditorError>
-              </DemoEditor>
-            )}
-          </Collapse>
+              {demoOptions.disableLiveEdit
+                ? tabs.map((tab, index) => (
+                    <DemoCodeViewer
+                      key={index}
+                      code={tab.raw}
```

Comment:
It feels like there is a mix of conditions for 
- tabs vs single block 
- editable vs preview

Would it make sense to always use the `TabPanel` and switch between editable/non-editable at the tab level? Something like

```js
tabs.map((tab, index) => 
  (demoOptions.disableLiveEdit || index > 0)
    ? <DemoCodeViewer ... />
     : <TabPanel> ... </TabPanel>
)
```

Reply:
Yes, did you mean: 

```js
tabs.map((tab, index) => 
  (demoOptions.disableLiveEdit || index > 0)
    ? <DemoCodeViewer ... />
     : <DemoEditor> ... </DemoEditor>
)
```